recipes/corrade/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+        "with_interconnect": [True, False],
+        "with_pluginmanager": [True, False],
+        "with_testsuite": [True, False],
+        "with_utility": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_deprecated": True,
+        "with_interconnect": True,
+        "with_pluginmanager": True,
+        "with_testsuite": True,
+        "with_utility": True,
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version.value) < 14:
+            raise ConanInvalidConfiguration("Corrade requires Visual Studio version 14 or greater")
+        if tools.cross_building(self.settings):
+             raise ConanInvalidConfiguration("This Corrade recipe doesn't support cross building yet")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_STATIC"] = "ON" if not self.options.shared else "OFF"
+        cmake.definitions["BUILD_DEPRECARED"] = "ON" if not self.options["build_deprecated"] else "OFF"
+        cmake.definitions["WITH_INTERCONNECT"] = "ON" if not self.options["with_interconnect"] else "OFF"
+        cmake.definitions["WITH_PLUGINMANAGER"] = "ON" if not self.options["with_pluginmanager"] else "OFF"
+        cmake.definitions["WITH_TESTSUITE"] = "ON" if not self.options["with_testsuite"] else "OFF"
+        cmake.definitions["WITH_UTILITY"] = "ON" if not self.options["with_utility"] else "OFF"
+
+        cmake.definitions["WITH_RC"] = "OFF"  # This executable goes to its own package corrade-rc
+
+        # Corrade uses suffix on the resulting 'lib'-folder when running cmake.install()
+        # Set it explicitly to empty, else Corrade might set it implicitly (eg. to "64")
+        cmake.definitions["LIB_SUFFIX"] = ""
+
+        if self.settings.compiler == "Visual Studio":
+            cmake.definitions["MSVC2015_COMPATIBILITY"] = "ON" if tools.Version(self.settings.compiler.version.value) == 14 else "OFF"
+            cmake.definitions["MSVC2017_COMPATIBILITY"] = "ON" if tools.Version(self.settings.compiler.version.value) == 14 else "OFF"
+
+        cmake.configure(build_folder=self._build_subfolder)
+
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, 'share')) # does only contain .cmake files
```

Comment:
Do you know if these files contain any kind of utility for package consumers (not find/config files) ?

Reply:
I'm not sure about the `shared/UseCorrade.cmake` file

@mosra Would you mind explaining if we should package this file for consumers of Corrade?

Reply:
Just look at the file ;) -- yes, it contains a ton of macros and helpers (`corrade_add_test()`, `corrade_add_resource()`, ...) that are useful for users.

Reply:
Please package them accordingly using the `cpp_info.buildidrs` and the new `cpp_info.build_modules`

Reply:
@Croydon did you try `cpp_info.build_modules` to let consumers reuse those CMake macros? I cannot see it in the recipe